ukvm/ukvm_module_net.c

```diff
@@ -109,13 +107,14 @@ static int tap_attach(const char *ifname)
         errno = ENOENT;
         return -1;
     }
-    if (!up) {
+
+#if defined(__linux__)
+
+    if (! (ifp->ifa_flags & (IFF_UP | IFF_RUNNING))) {
```

Comment:
This is a use-after-free, `ifa` (and thus `ifp`) is freed on line 105 (or 107 in the original).

I suggest to keep the original structure (collecting `up` inside the for loop) and just make the `if (!up)` conditional on `__linux__`.